firefox-ios/EcosiaTests/Account/Auth/AuthNotificationSystemTests.swift

```diff
@@ -0,0 +1,460 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import XCTest
+import Common
+@testable import Ecosia
+
+final class AuthNotificationSystemTests: XCTestCase {
+
+    var authStateManager: AuthStateManager!
+    var windowRegistry: EcosiaAuthWindowRegistry!
+    var testWindowUUID: WindowUUID!
+    fileprivate var notificationObserver: NotificationObserver!
+
+    override func setUp() {
+        super.setUp()
+        authStateManager = AuthStateManager.shared
+        windowRegistry = EcosiaAuthWindowRegistry.shared
+        testWindowUUID = WindowUUID.XCTestDefaultUUID
+        notificationObserver = NotificationObserver()
+        
+        // Clear state management system
+        authStateManager.clearAllStates()
+        windowRegistry.clearAllWindows()
+        
+        // Register a test window
+        windowRegistry.registerWindow(testWindowUUID)
+    }
+
+    override func tearDown() {
+        // Clean up state after each test
+        authStateManager.clearAllStates()
+        windowRegistry.clearAllWindows()
+        notificationObserver.cleanup()
+        notificationObserver = nil
+        authStateManager = nil
+        windowRegistry = nil
+        testWindowUUID = nil
+        super.tearDown()
+    }
+
+    // MARK: - Notification Names Tests
+
+    func testNotificationNames_haveCorrectValues() {
+        // Test all notification names have correct string values
+        XCTAssertEqual(Notification.Name.EcosiaAuthStateChanged.rawValue, "EcosiaAuthStateChanged")
+        XCTAssertEqual(Notification.Name.EcosiaAuthDidLoginWithSessionToken.rawValue, "EcosiaAuthDidLoginWithSessionToken")
+        XCTAssertEqual(Notification.Name.EcosiaAuthDidLogout.rawValue, "EcosiaAuthDidLogout")
+        XCTAssertEqual(Notification.Name.EcosiaAuthStateReady.rawValue, "EcosiaAuthStateReady")
+        XCTAssertEqual(Notification.Name.EcosiaAuthShouldLogoutFromWeb.rawValue, "EcosiaAuthShouldLogoutFromWeb")
+    }
```

Comment:
This seems like just testing values inside constants, do you see value in it?

Reply:
At a glance, I would say no, but a typo on such notifications might be done (and slip off the reviews).
In case within a refactor we happen to change such values (or simply accidentally insert a typo), we'd catch that.

Reply:
But if a name changes, I guess everything will still work as intended given all parts consume from the same constants, won't it? I'd expect that even if something does not work that the other tests catch it then.

Even so, I understand that this test would act as a 2FA for those exact names, so no issue if you want to keep them (even if no functionality is affected - unless I'm missing something).